cmd/scaffold.go

```diff
@@ -188,16 +188,16 @@ func (fnb *FlowNodeBuilder) enqueueNetworkInit() {
 		nodeRole := nodeID.Role
 
 		var nodeTopology topology.Topology
-		if nodeRole == flow.RoleCollection {
-			nodeTopology, err = topology.NewCollectionTopology(nodeID.NodeID, fnb.State)
-		} else {
-			nodeTopology, err = topology.NewRandPermTopology(nodeRole, nodeID.NodeID)
-		}
+
+		nodeTopology, err = topology.NewTopicAwareTopology(fnb.Me.NodeID(), fnb.State)
+
```

Comment:
Per conversation with @vishalchangrani it was decided to let having solely a single topology for all types of nodes that operates based on "topics" the nodes involved in, and not necessarily their roles.